src/validators/datetime.rs

```diff
@@ -65,9 +65,12 @@ impl Validator for DateTimeValidator {
         state: &mut ValidationState,
     ) -> ValResult<PyObject> {
         let strict = state.strict_or(self.strict);
-        let datetime = input
-            .validate_datetime(strict, self.microseconds_precision)?
-            .unpack(state);
+        let datetime = match input.validate_datetime(strict, self.microseconds_precision) {
+            Ok(val_match) => val_match.unpack(state),
+            // if the error was a parsing error, in lax mode we allow dates and add the time 00:00:00
+            Err(line_errors @ ValError::LineErrors(..)) if !strict => datetime_from_date(input)?.ok_or(line_errors)?,
```

Comment:
I suspect that here we want to call `state.floor_exactness(Exactness::Lax)` given this is a lax conversion.